lib/msf/core/post/file.rb

```diff
@@ -501,19 +501,25 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    unless exist?(old_file)
+      elog("The file to be renamed does not exists!")
+      return false
+    end
```

Comment:
Have you considered possibly removing this? It looks like each of the commands handles the failure conditions correctly. The Meterpreter one will raise an exception, but that could be caught, logged and noramlized to `false` so the API is still consistent.

If we can remove this, we'd reduce this operation to one command that would need to be run which would speed it up quite a bit.

Reply:
Cool will update this! :+1: